### PR TITLE
Fix Mac and Windows compilation error

### PR DESCRIPTION
**Pull request by moninom1**
Description
-----------
The build combination test was failing on Mac as we were compiling Windows port which uses some assembly instructions were not available on Mac. It also failed on Windows most likely because it was never built on Windows. 

This PR changes the build to use POSIX port on Mac and Linux and Windows port on Windows platform. It also removes stubs which are not needed anymore. 

Test Steps
-----------
Test Command - 
```
cmake -S test/build-combination -B test/build-combination/build/  -DTEST_CONFIGURATION=ENABLE_ALL
make -C test/build-combination/build/
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by aggarg — DISMISSED** [review]

File: test/build-combination/CMakeLists.txt [hunk 1]
```diff
@@ -41,7 +41,7 @@ set( TEST_DIR ${MODULE_ROOT_DIR}/test/build-combination )
 include_directories( ${MODULE_ROOT_DIR}/source/include )
 include_directories( ${MODULE_ROOT_DIR}/source/portable/Compiler/MSVC )
 include_directories( ${FREERTOS_KERNEL_DIR}/include )
-include_directories( ${FREERTOS_KERNEL_DIR}/portable/MSVC-MingW )
+include_directories( ${FREERTOS_KERNEL_DIR}/portable/ThirdParty/GCC/Posix )
```
Comment (AniruddhaKanhere) [inline test/build-combination/CMakeLists.txt:44]: Does this change render the build useless on windows based platforms?

File: test/build-combination/CMakeLists.txt [hunk 2]
```diff
@@ -67,3 +67,7 @@ add_executable(project ${KERNEL_SOURCES}
 		       ${FREERTOS_KERNEL_DIR}/portable/MemMang/heap_4.c
 		       ${MODULE_ROOT_DIR}/source/portable/BufferManagement/BufferAllocation_2.c
 		       ${TEST_DIR}/Common/main.c )
+
+# Link pthread which is needed for POSIX port.
+find_package( Threads REQUIRED )
+target_link_libraries( project Threads::Threads )
```
Comment (AniruddhaKanhere) [inline test/build-combination/CMakeLists.txt:73]: What about the systems which do not have these packages?

File: test/build-combination/CMakeLists.txt (hunk 1 above)
Comment (aggarg) [reply test/build-combination/CMakeLists.txt:44]: Was it designed to be used on Windows? If so, should we not use Windows port on Windows and POSIX port on UNIX platforms?

File: test/build-combination/CMakeLists.txt (hunk 2 above)
Comment (aggarg) [reply test/build-combination/CMakeLists.txt:73]: POSIX port requires pthread - so a reasonable approach would be to fail?

File: test/build-combination/CMakeLists.txt (hunk 1 above)
Comment (AniruddhaKanhere) [reply test/build-combination/CMakeLists.txt:44]: It is just a build check. It doesn't create a working application. It just makes sure that in the combinations there is no compilation error.
If at all possible, I would like if we can include the directories conditionally based on the underlying OS.

File: test/build-combination/CMakeLists.txt (hunk 2 above)
Comment (AniruddhaKanhere) [reply test/build-combination/CMakeLists.txt:73]: Or, we can include the POSIX specific stuff too.

**Comment by AniruddhaKanhere** [comment]
An overarching comment @aggarg - these build checks were first written for the CI which uses Ubuntu.
But, since we are changing this to work with MAC too, I was thinking of adding in changes which would enable this to be compiled on windows as well. I understand if we would like to do that in other PR - in that case, I am happy to approve the PR.

**Review by AniruddhaKanhere — APPROVED** [review]

**Comment by AniruddhaKanhere** [comment]
Thank you for the changes @aggarg and @moninom1!
I approve of this PR :)

**Review by aggarg — APPROVED** [review]

**Pull request merged**